wire-ios/Wire-iOS/Sources/UserInterface/App Lock/AppLockModule.Presenter.swift

```diff
@@ -67,12 +67,10 @@ extension AppLockModule.Presenter: AppLockPresenterViewInterface {
 
     func processEvent(_ event: AppLockModule.Event) {
         switch event {
-        // In iOS 14, it was found that 'viewDidAppear' may be invoked even when the app is in the background. To prevent re-authentication when the app is in the background, there is the 'requireActiveApp' parameter.
-        case .viewDidAppear, .unlockButtonTapped:
-            interactor.executeRequest(.initiateAuthentication(requireActiveApp: true))
-
-        case .applicationWillEnterForeground:
-            interactor.executeRequest(.initiateAuthentication(requireActiveApp: false))
+        // In iOS 14, it was found that 'viewDidAppear' may be invoked even when the app is in the background.
```

Comment:
I guess as we don't support iOS14 it's not relevant anymore, were you able to confirm it's fixed on newest iOS versions?

Reply:
My assumption based on when the comment was written is that it became broken in iOS 14. I have tested my fix on iOS 18 device and iOS 17.5 simulator. I _think_ my fix would also keep working on iOS 14. However, I think it needs a good round of QA

Reply:
I will test myself on other sim versions (iOS 15 etc) soon, before merging any cherrypick into the release branch

Reply:
I have now tested on a range of simulators down to iOS 15.2 without issue. iOS 15.0 simulator doesn't work at all on my system but this is not related to the app.